qiskit_experiments/data_processing/data_processor.py

```diff
@@ -97,20 +105,23 @@ def call_with_history(
             processed data: The datum processed by the data processor.
             history: The datum processed at each node of the data processor.
         """
-        return self._call_internal(datum, True, history_nodes)
+        if not self.is_trained:
+            self.train(data)
```

Comment:
Do we still want to keep this logic? Seems like a series of nodes before the node to be trained are called twice for training and processing. This may hurt performance when we process a job result with huge number of entries.

Reply:
Good point. Its probably best to do without this for now especially since `curve_analysis` calls the training step. See 9c95db6. If we want something more sophisticated I would make a dedicated PR.